File: Assignment42/Question2.c
#include <stdio.h>

int Sum(int iNo) 
{
    if (iNo == 0) 
    {
        return 0;
    } 
    else 
    {
        return (iNo % 10) + Sum(iNo / 10);
    }
}

int main() 
{
    int iValue = 0;
    int iRet = 0;

    printf("Enter The Number: ");
    scanf("%d", &iValue);

    iRet = Sum(iValue);

    printf("Sum of digits is: %d\n", iRet);

    return 0;
}

File: Assignment54/Question2.java
import java.util.Scanner;

class MyArray
{
    public void SumArray(int arr[])
    {
        // Traverse array
        for(int i = 0; i < arr.length; i = i + 1)
        {
            int num = arr[i];
            int sum = 0;

            // Summation of digits
            while(num != 0)
            {
                int digit = num % 10;
                sum = sum + digit;
                num = num / 10;
            }

            // Replace original element with summation of its digits
            arr[i] = sum;
        }
    }
}

class Question2
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        System.out.println("Enter number of elements : ");
        int size = sobj.nextInt();

        int Arr[] = new int[size];

        System.out.println("Enter elements : ");
        for(int i = 0; i < Arr.length; i = i + 1)
        {
            Arr[i] = sobj.nextInt();
        }

        MyArray mobj = new MyArray();
        mobj.SumArray(Arr);

        System.out.println("Array after replacing each number with sum of digits : ");
        for(int i = 0; i < Arr.length; i = i + 1)
        {
            System.out.print(Arr[i] + " ");
        }

        Arr = null;
    }
}